### PR TITLE
fix(i18n): Fix bug where messages starting with HTML cause an exception

### PR DESCRIPTION
**Pull request by ehoogerbeets**
A message like this:

```jsx
<FormattedCompMessage
  description="Body for tooltip pointing to add user button"
    id="adminConsole.components.Onboarding.addUserButtonTooltipBodyText"
>
    <b>Managed users</b> are employees in your Box Enterprise. They often share your company domain(s) and require a greater amount of control, while <b>External users</b> are clients or partners who have been invited to collaborate on specific content in your enterprise.
</FormattedCompMessage>
```

Caused an exception even though it is perfectly valid.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @ehoogerbeets has signed the CLA. Thanks for the pull request!

File: i18n/en-US.properties [hunk 1]
```diff
@@ -208,6 +208,8 @@ be.deleteDialogFileText = Are you sure you want to delete {name}?
 be.deleteDialogFolderText = Are you sure you want to delete {name} and all its contents?
 # Label for delete confirmation dialog
 be.deleteDialogLabel = Confirm Delete
+# Aria label for button to delete a comment, task, or app activity
+be.deleteLabel = Delete
```
Comment (jstoffan) [inline i18n/en-US.properties]: Are these changes related to this PR?

File: src/components/i18n/Composition.js [hunk 2]
```diff
@@ -64,6 +65,15 @@ class Composition {
         return this.ma.getMinimalString();
     }
 
+    /**
+     * @private
+     */
+    nextKey() {
+        const ret = `key${this.keyIndex}`;
```
Comment (jstoffan) [inline src/components/i18n/Composition.js]: We usually try to avoid abbreviations, where possible, as they can hinder readability.

File: src/components/i18n/FormattedCompMessage.js [hunk 3]
```diff
@@ -150,7 +150,7 @@ class FormattedCompMessage extends React.Component<Props, State> {
     }
 
     render() {
-        const { count, tagName, intl, description, id, ...rest } = this.props;
+        const { count, tagName, intl, description, id, defaultMessage, ...rest } = this.props;
```
Comment (jstoffan) [inline src/components/i18n/FormattedCompMessage.js]: It looks like `defaultMessage` is unused. Is that intentional?

File: i18n/en-US.properties (hunk 1 above)
Comment (ehoogerbeets) [reply i18n/en-US.properties]: No, git push automatically put them there

File: src/components/i18n/FormattedCompMessage.js (hunk 3 above)
Comment (ehoogerbeets) [reply src/components/i18n/FormattedCompMessage.js]: Yes, it was getting into the "rest" which made its way into the output. Anything that is not a prop we use directly should go onto the output span element like className for example.

File: i18n/en-US.properties (hunk 1 above)
Comment (ehoogerbeets) [reply i18n/en-US.properties]: I just did a rebase onto the latest from master, so I am not sure why that is happening...

File: src/components/i18n/Composition.js (hunk 2 above)
Comment (ehoogerbeets) [reply src/components/i18n/Composition.js]: You would prefer something like "returnValue" perhaps?

Comment (jstoffan) [reply src/components/i18n/Composition.js]: There's always good ol' `result`.

File: i18n/en-US.properties (hunk 1 above)
Comment (jstoffan) [reply i18n/en-US.properties]: Try deleting your `/es` directory, then run `./scripts/prepush.sh`.

Comment (ehoogerbeets) [reply i18n/en-US.properties]: Yeah, talked with Andrew H and he essentially said the same thing... someone deleted some strings that still existed in my es dir. He recommended to do "yarn clean" and "yarn build" after every rebase. Now the en-US.properties is gone from the PR.

File: src/components/i18n/Composition.js (hunk 2 above)
Comment (ehoogerbeets) [reply src/components/i18n/Composition.js]: Changed to "result"

File: i18n/en-US.properties (hunk 1 above)
Comment (booc0mtaco) [reply i18n/en-US.properties]: to be clear, this can occur if there are stale values in `i18n/json` and a rebased commit contains a deletion to some content referenced there
